main.go

```diff
@@ -550,26 +575,41 @@ func DownloadS3Backup(c *cli.Context) error {
 			"s3-accessKey":  bc.AccessKey,
 			"s3-region":     bc.Region,
 		}).Errorf("failed to set s3 server: %s", err)
-		return fmt.Errorf("faield to set s3 server: %+v", err)
+		return fmt.Errorf("failed to set s3 server: %+v", err)
 	}
 
 	fileName := c.String("name")
+	compressedFileName := fmt.Sprintf("%s.%s", fileName, compressedExtension)
 	if len(fileName) == 0 {
 		return fmt.Errorf("empty file name")
 	}
 
-	log.Infof("invoking downloading backup files: %s", fileName)
-	for retries := 0; retries <= s3ServerRetries; retries++ {
-		err := client.FGetObject(bc.BucketName, fileName, backupBaseDir+"/"+fileName, minio.GetObjectOptions{})
-		if err != nil {
-			log.Infof("failed to download etcd snapshot file: %v, retried %d times", err, retries)
-			if retries >= s3ServerRetries {
-				return fmt.Errorf("failed to download etcd snapshot file: %v", err)
+	fileNameOptions := [2]string{compressedFileName, fileName}
+	for _, fileNameOption := range fileNameOptions {
+		log.Infof("invoking downloading backup files: %s", fileNameOption)
+		for retries := 0; retries <= s3ServerRetries; retries++ {
+			err := client.FGetObject(bc.BucketName, fileNameOption, backupBaseDir+"/"+fileNameOption, minio.GetObjectOptions{})
+			if err != nil {
+				log.Infof("failed to download etcd snapshot file [%s]: %v, retried %d times", fileNameOption, err, retries)
+				if retries >= s3ServerRetries {
+					log.Warningf("failed to download etcd snapshot file [%s]: %v", fileNameOption, err)
+					break
```

Comment:
should we return err if it fails for all options?

Reply:
Updated the PR, thanks